src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/TransfersController.cs

```diff
@@ -100,4 +102,117 @@ public async Task<ActionResult<ResultList<Transfer>>> GetTransfers([FromServices
                 })
         };
     }
+
+    /// <summary>
+    /// Queues a request to transfer a certificate to another wallet for the authenticated user.
+    /// </summary>
+    /// <param name="bus"></param>
+    /// <param name="request"></param>
+    /// <response code="202">Transfer request has been queued for processing.</response>
+    /// <response code="401">If the user is not authenticated.</response>
+    [HttpPost]
+    [Route("v1/transfers")]
+    [Produces("application/json")]
+    [ProducesResponseType(typeof(TransferResponse), StatusCodes.Status202Accepted)]
+    [ProducesResponseType(typeof(void), StatusCodes.Status401Unauthorized)]
+    public async Task<ActionResult<TransferResponse>> TransferCertificate(
+        [FromServices] IBus bus,
+        [FromBody] TransferRequest request
+    )
+    {
+        if (!User.TryGetSubject(out var subject)) return Unauthorized();
+
+        var command = new TransferCertificateCommand
+        {
+            TransferRequestId = Guid.NewGuid(),
+            Owner = subject,
+            Registry = request.CertificateId.Registry,
+            CertificateId = request.CertificateId.StreamId,
+            Quantity = request.Quantity,
+            Receiver = request.ReceiverId,
+            HashedAttributes = request.HashedAttributes,
+        };
+
+        await bus.Publish(command);
```

Comment:
I don't believe this should be awaited?

Reply:
It should, to make sure it happens on the same task and before the response is returns to the caller.
Otherwise it could fail in a separate task.

Reply:
Maybe we should talk about this over teams